scalyr_agent/builtin_monitors/redis_monitor.py

```diff
@@ -206,7 +207,11 @@ def log_entry( self, logger, entry ):
         except UnicodeDecodeError, e:
             if self.utf8_warning_interval:
                 logger.warn( "Redis command contains invalid utf8: %s" % binascii.hexlify( entry['command'] ), limit_once_per_x_secs=self.utf8_warning_interval, limit_key="redis-utf8" )
-            command = entry['command'].decode( 'utf8', errors="replace" )
+            command = entry['command'].decode('utf8', errors="replace")
+            # echee: The following replacement should work as a pre 2.7
+            # replacement for the above line but I'm not comfortable activating
+            # it until we have a unit test given that it could mess up REDIS
+            # command = codecs.decode(entry['command'], 'utf8', 'replace')
```

Comment:
I _think_ this alternative line is equivalent to the existing (line 210) line which is not python 2.6 compatible because of the kwarg.  However I'm nervous about including this fix until confident that we have test coverage.  WDYT?

Reply:
I'm not sure the alternative line works.  According to the [2.6 docs](https://docs.python.org/2.6/library/codecs.html#codecs.Codec.decode), the encoding is not specified as a parameter, rather it appears you need to manually get the decoder for a specific encoding from a factory function [`get_decoder`](https://docs.python.org/2/library/codecs.html#codecs.getdecoder).

I'll do some manual testing shortly, but we may need to look at writing our own wrapper shims that get loaded depending on the version.

Also, is this the only place we use `decode`?  I think it appears in other places in the code also.

Reply:
I think you're right that it doesn't work.  I must have been mistakenly looking at 2.7 `Codec` documentation.  And yes we'll have to fix elsewhere.  (I think the unit test suite just exercises this one and the others might not be covered.)